### PR TITLE
[Feature][seatunnel-connector] Add tidb spark source

### PR DESCRIPTION
**Pull request by zixi0825**
## Purpose of this pull request

 This pull request adds tidb spark source.#913

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [x] Change does not need document change, or I will submit document change to https://github.com/apache/incubator-seatunnel-website later

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -115,6 +115,7 @@
         <docker.tag>${project.version}</docker.tag>
         <jcommander.version>1.81</jcommander.version>
         <junit.version>4.13.2</junit.version>
+        <tispark.version>2.1.9-spark_2.4</tispark.version>
```
Comment (RickyHuo) [inline pom.xml:118]: I think Tispark 2.4.x would be better.

**Review by RickyHuo — CHANGES_REQUESTED** [review]

File: docs/en/configuration/source-plugins/Tidb.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+### spark.sql.extensions [string]
+
+default value : org.apache.spark.sql.TiExtensions
+
+## Options
+
+| name           | type   | required | default value |
+| -------------- | ------ | -------- | ------------- |
+| database       | string | yes      | -             |
```
Comment (RickyHuo) [inline docs/en/configuration/source-plugins/Tidb.md:26]: The docs markdown should be like demo below to support redirect.
```
| database(#database-string)     |  string | yes      | -             |
```

File: seatunnel-connectors/seatunnel-connector-spark-tidb/src/main/scala/org/apache/seatunnel/spark/source/Tidb.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  override def getData(env: SparkEnvironment): Dataset[Row] = {
+    val database = config.getString("database")
+
+    val ti = new TiContext(env.getSparkSession)
+    ti.tidbMapDatabase(database)
+    env.getSparkSession.sql(config.getString("pre_sql"))
+  }
+}
```
Comment (RickyHuo) [inline seatunnel-connectors/seatunnel-connector-spark-tidb/src/main/scala/org/apache/seatunnel/spark/source/Tidb.scala:43]: File must end with a newline character.

File: pom.xml (hunk 1 above)
Comment (zixi0825) [reply pom.xml]: change version to 2.4.1

**Comment by zixi0825** [comment]
@RickyHuo @CalvinKirs

**Comment by RickyHuo** [comment]
@zixi0825  According to https://github.com/pingcap/tispark

> For TiSpark version >= 2.0:

```
spark.sql("use tpch_test")
spark.sql("select count(*) from lineitem").show
```

Have you check with current version ?

**Comment by zixi0825** [comment]
> @zixi0825 According to https://github.com/pingcap/tispark
> 
> > For TiSpark version >= 2.0:
> 
> ```
> spark.sql("use tpch_test")
> spark.sql("select count(*) from lineitem").show
> ```
> 
> Have you check with current version ?

It's my mistake. I'll recheck and modify it

**Comment by RickyHuo** [comment]
The directory of doc seem error with branch of dev.

**Comment by RickyHuo** [comment]
@zixi0825 Please resolve the conflicts.

**Comment by zixi0825** [comment]
@RickyHuo

**Review by RickyHuo — APPROVED** [review]
LGTM

**Pull request merged**